### PR TITLE
Don't use lld on windows

### PR DESCRIPTION
**Pull request by martinthomson**
The windows build ignores this attribute, but there is no point in having a warning.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -76,7 +77,7 @@ jobs:
         uses: ilammy/msvc-dev-cmd@v1
 
       - name: Install Rust tools
-        run: cargo +${{ matrix.rust-toolchain }} binstall --no-confirm cargo-llvm-cov cargo-nextest
+        run: cargo +${{ matrix.rust-toolchain }} binstall --no-confirm bench cargo-llvm-cov cargo-nextest
```
Comment (larseggert) [inline .github/workflows/check.yml:80]: Did this sneak in from another PR?

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -114,6 +115,9 @@ jobs:
       - name: Run tests and determine coverage
         run: cargo +${{ matrix.rust-toolchain }} llvm-cov nextest --features ci --all-targets --no-fail-fast --lcov --output-path lcov.info
 
+      - name: Benches should at least build
+        run: cargo +${{ matrix.rust-toolchain }} bench --features bench --no-run
+
```
Comment (larseggert) [inline .github/workflows/check.yml:120]: Ditto?

**Review by larseggert — COMMENTED** [review]
Did something leak in from another PR?

**Comment by larseggert** [comment]
Also see https://www.reddit.com/r/rust/comments/nivznh/using_lld_as_the_linker_on_windows/ for how to use lld on Windows. (It should be installed in the runner somewhere: https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md)

File: .github/workflows/check.yml (hunk 2 above)
Comment (martinthomson) [reply .github/workflows/check.yml]: Yeah, I'm trying to get benches built, in anticipation of one of the benchmarks landing.  Though I can move that to the other PR.

Comment (larseggert) [reply .github/workflows/check.yml]: No, leave it in. I just wanted to check that it's deliberate.

**Review by larseggert — APPROVED** [review]

**Pull request merged**